File: Client/src/app/_generated/models/warehouse-address-response.ts
/* tslint:disable */
import { WarehouseCityResponse } from './warehouse-city-response';

/**
 * Response dto for adress data inside a request targeting the warehouse controller
 */
export interface WarehouseAddressResponse {
  id?: string;

  /**
   * The street name without the house-number
   */
  street?: string;

  /**
   * The housenumber , can be null if non existing
   */
  housenumber?: number;
  cityId?: string;
  city?: WarehouseCityResponse;
}

File: API/API/Features/v1/WarehouseArea/WarehouseAddressResponse.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CLERP.API.Features.v1.WarehouseArea
{
    /// <summary>
    /// Response dto for adress data inside a request targeting the warehouse controller
    /// </summary>
    public class WarehouseAddressResponse
    {
        [JsonProperty("id")]
        public Guid AddressGuid { get; set; }

        /// <summary>
        /// The street name without the house-number
        /// </summary>
        public string Street { get; set; }

        /// <summary>
        /// The housenumber , can be null if non existing
        /// </summary>
        [JsonProperty(NullValueHandling = NullValueHandling.Include, PropertyName = "housenumber")]
        public int? Housenumber { get; set; }

        [JsonProperty("cityId")]
        public Guid CityGuid { get; set; }

        [JsonProperty("city")]
        public WarehouseCityResponse City { get; set; }
    }
}
